int8 id
string state
bool collision
---
bool success